Adding custom report filter (#163)
SkyAPM/go2sky: README.md

}
defer r.Close()
tracer, err := go2sky.NewTracer("example", go2sky.WithReporter(r))
```

> In some scenarios, we may need a filter to filter segments that do not need to be submitted to reduce the pressure of grpc reporting. For example, we just need to track the request of error.

Reviewer: ```suggestion
In some scenarios, we may need a filter to filter segments that do not need to be submitted, for example, to reduce the load of gRPC reporting, or only track the request of error.
```
